marketplace/deployer_util/expand_config.py

```diff
@@ -73,6 +77,14 @@ def expand(values_dict, schema):
       result[k] = generate_password(prop.password)
       continue
 
+    if v is None and prop.xtype == 'APPLICATION_UID':
+      if not app_uid:
+        raise InvalidProperty(
+            'Property {} is of type APPLICATION_UID, but no value was specified'
```

Comment:
"no value" is misleading--change to "--app_uid was not specified".

There is a separate question whether we should respect a directly specified value. I think always overriding it is fine for now.

Reply:
Done!